lib.php

```diff
@@ -207,3 +198,29 @@ function tool_crawler_extend_navigation_course($navigation, $course, $coursecont
         }
     }
 }
+/**
+ * Add a batch of ad hoc crawl tasks
+ * @param integer $limitadhoctasks the limit of concurrent adhoc crawl tasks
+ * @param boolean $verbose show verbose feedback
+ */
+function tool_crawler_add_adhoc_task($limitadhoctasks, $verbose = false) {
+    // Get number of adhoc crawl tasks we currently have.
+    $currentadhoc = tool_crawler_get_adhoc_tasks();
```

Comment:
This doesn't sit perfectly well with me because we are reaching into the private internals of the Task API.

A better alternative might be to create N tasks, each with custom data of 'worker = 1', 'worker = 2', etc and then tell the Task API to ignroe already queued duplicates:

https://docs.moodle.org/dev/Task_API#Ignore_duplicate_adhoc_tasks

Reply:
Sounds sensible, this has been updated now. Tested that if we have the same worker customdata existing, then it won't create that adhoc task again.